tests/lib/rules/use-waiters-in-module-scope.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const rule = require('../../../lib/rules/use-waiters-in-module-scope');
+
+const { ERROR_MESSAGE } = rule;
+const RuleTester = require('eslint').RuleTester;
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('use-waiters-in-module-scope', rule, {
+  valid: [
+    `
+    import { buildWaiter } from 'ember-test-waiters';
+
+    let myWaiter = buildWaiter('waiterName');
+  `,
+    `
+    import { buildWaiter } from 'table-waiters';
+
+    function useWaiter() {
+      let myOtherWaiter = buildWaiter('the second');
+    }
+  `,
```

Comment:
There should be valid test case:

```
import { buildWaiter } from 'ember-test-waiters';

buildWaiter('waiterName');
```

Reply:
You would never invoke it this way. You always get a reference to the return value of `buildWaiter`. We don't provide a public API that we want people to use to retrieve waiters any other way. Additionally, there's an important fact of waiters that they should be used in proximity to the code they are waiting on. I've fixed many, many hard to understand issues that were the result of people building highly convoluted waiter setups. Simple is definitely better, in this case. 

I plan on massively extending the documentation and guidance around waiters in that repo over the next few weeks, and will include this vital information there.

To speak specifically to the case above, I would actually consider this invalid. Perhaps I should make a test case for it? cc @rwjblue

Reply:
My interest is actually just in ensuring that the rule does not crash as somebody is typing out their code. That's why we need test cases to cover variations like this, not because these test cases are a valid way of writing the code.

Reply:
OK, very fair point. You're just looking to add resilience. That said if I make the rule pass on bad patterns, that's not really doing its job. I'd prefer to extend the rule to indicate multiple negative patterns, such as this (you always want to store the returned reference of the waiter). Thoughts on this? If I go this route, I'll likely widen the name to encompass more of waiter best practices.

Reply:
Something like: `require-test-waiters-valid-usage`. You like?

Reply:
Regarding checking that the return value of `buildWaiter` is used, is this a common mistake developers make? There are probably hundreds of functions we could add such a check for, and I’m generally not sure it’s worth adding lint rules to ensure developers are using the return values of specific function calls, unless there’s a specific reason to.

As for the question about having separate rules or a consolidated rule: I would use separate rules to catch unrelated bad practices to keep each rule short and focused and give consumers more flexibility, while I would use a consolidated rule (perhaps called `no-invalid-test-waiters`) for checks that are related or convenient to add together.

Reply:
Using the returned, built waiter from `buildWaiter` is entirely the way this addon is expected to be used. You create a waiter, and ultimately use it to signal the beginning and ending of async behavior. Calling `buildWaiter` on its own, without doing anything with the return value doesn't do anything. 

Due to the narrow recommended usage of this API, I think it's entirely reasonable to ensure a few things:
1. `buildWaiter` is invoked in module scope
1. The return value of `buildWaiter` is stored in a variable to be used in that module

Not doing either of these things could be considered wrong, and we should likely guard against both.

Reply:
Sounds good to me.

Reply:
Updated.